### PR TITLE
Add discounts info for paypal payments

### PR DESCRIPTION
**Pull request by krzysztofwolski**
Add option to add discounts info to PayPal payments.

todo:

- [x] Docs
- [x] Tests

### CODE REVIEW

File: payments/paypal/__init__.py [hunk 1]
```diff
@@ -171,8 +171,19 @@ def get_transactions_items(self, payment):
                     'sku': purchased_item.sku}
             yield item
 
+    def get_discounts(self, payment):
+        for discount in payment.get_discounts():
+            price = -1 * discount.amount
```
Comment (patrys) [inline payments/paypal/__init__.py]: `price = -discount.amount`?

File: payments/__init__.py [hunk 2]
```diff
@@ -10,6 +10,9 @@
 PurchasedItem = namedtuple('PurchasedItem',
                            'name, quantity, price, currency, sku')
 
+Discount = namedtuple('Discount',
+                      'name, quantity, amount, currency')
```
Comment (patrys) [inline payments/__init__.py]: Do we need quantity for discounts?

**Comment by patrys** [comment]
We can't have the discount docs specific to PayPal. The point of offering one API is so that the code does not have to know it's dealing specifically with PayPal.

**Comment by krzysztofwolski** [comment]
We decited to not change api for only one payment provider. Also, adding discount in product list as item with negative price is suggested solution by PayPal docs.

**Pull request closed without merging**